python/mpt/src/mpt/trie_diff.cairo

```diff
@@ -443,33 +450,34 @@ func _process_storage_diff{
     }
 
     if (left.value != 0) {
-        let left_decoded = U256_from_rlp(left.value.value.value.bytes);
+        let left_u256 = U256_from_rlp(left.value.value.value.bytes);
         tempvar range_check_ptr = range_check_ptr;
         tempvar bitwise_ptr = bitwise_ptr;
         tempvar poseidon_ptr = poseidon_ptr;
         tempvar keccak_ptr = keccak_ptr;
     } else {
-        tempvar left = left;
+        // If the left doesn't have a value, it means the storage key was deleted, the new value is 0.
+        tempvar left_u256 = U256(cast(u256_zero, U256Struct*));
```

Comment:
based on how I understand things, the state diff will also have a 0 there, so I am surprised this is even needed (note to self). 

There might be an issue with how the `ids` object is built, leading to a slight mishap in how this value was represented in the debug data...